src/Query/Terms.php

```diff
@@ -56,8 +56,10 @@ public function setTerms(array $terms): self
 
     /**
      * Adds a single term to the list.
+     *
+     * @param float|int|string $term
      */
-    public function addTerm(string $term): self
+    public function addTerm($term): self
     {
         $this->terms[] = $term;
```

Comment:
Would probably be judicious to throw an error when type is not accepted? That would make the type-hint change when supporting PHP 8.0 as minimal version less impacting, as code passing an invalid type would already be breaking.

```suggestion
        if (!\is_string($term) && !\is_int($term) && !\is_float($term)) {
            throw new \TypeError(\sprintf('Argument 1 passed to "%s()" must be of type float|int|string, %s given.', __METHOD__, \is_object($term) ? \get_class($term) : \gettype($term)));
        }
        
        $this->terms[] = $term;
```

Reply:
Applied!